versioned/tiered/rocksdb/src/main/java/com/dremio/nessie/versioned/store/rocksdb/RocksDBStore.java

```diff
@@ -224,7 +224,30 @@ public void load(LoadStep loadstep) throws NotFoundException {
 
   @Override
   public <C extends BaseValue<C>> boolean delete(ValueType<C> type, Id id, Optional<ConditionExpression> condition) {
-    throw new UnsupportedOperationException();
+    final ColumnFamilyHandle columnFamilyHandle = getColumnFamilyHandle(type);
+
+    try (final Transaction transaction = transactionDB.beginTransaction(new WriteOptions(), new OptimisticTransactionOptions())) {
+      final byte[] key = transaction.getForUpdate(new ReadOptions(), columnFamilyHandle, id.toBytes(), true);
+      if (key != null) {
+        if (condition.isPresent()) {
+          final RocksBaseValue<C> consumer = RocksSerDe.getConsumer(type);
+          // TODO: Does the critical section need to be locked?
+          // TODO: Critical Section - evaluating the condition expression and deleting the entity.
+          // Check if condition expression is valid.
+          if (!(((Evaluator) consumer).evaluate(condition.get().accept(ROCKS_DB_CONDITION_EXPRESSION_VISITOR)))) {
```

Comment:
Before you can evaluate the consumer you need to populate it will a call to 
RocksSerDe.deserializeToConsumer(type, key, consumer);

Otherwise consumer is empty which seems to be what is happening.

Reply:
Thanks Steve. I've addressed the issue in the latest commit.

Reply:
Looks good now.

Reply:
I don't think you need to cast to Evaluator.

Reply:
Done